### DISCUSSION TITLE
Best practice for root-level state organization in slices

### DISCUSSION THREAD

**Discussion by thepuzzlemaster** [reduxjs/redux-toolkit#1524] [Q&A] [answered]
I had initially set up my slices to look like this:
```
mySlice = {
  data: {
    // my data properties go here
    someProperty: someValue,
  },
  internalData: {
    // some additional properties here, which are only used internally (not pushed to db)
  }
}
```

I recently did a huge refactor where I removed the `data` object on all my slices, so it now looks like this:
```
mySlice = {
  // my data properties are now at the root level of this slice
  someProperty: someValue,
  internalData: {
    // I still keep this for storing internally used properties.
  }
}
```

I was thinking it would be a bad practice to nest all my main properties under a `data` object, but now I'm not so sure it makes any difference. While, my code is a little bit cleaner not having to reference something as `mySlice.data.myProperty`, I'm not stoked about the main properties being siblings of the `internalData` property. Previously, I could just reference `mySlice.data` and get all my real data.

So I suppose my question is, does it make a difference if I keep the majority of my state nested under a `data` property vs being at the root level of my slices? I tried looking for some guidance here, but I didn't really find anything that pointed me in one direction or the other.

Thanks for any guidance. Hopefully this makes sense. I can try to provide a more concrete example if not.

**Comment by markerikson** [comment, marked as answer]
Your actual state structure is entirely up to you.  That said, I've never seen a `data/internalData` sub-naming pattern used before.

**Reply by thepuzzlemaster** [reply]
Okay thanks. I think I may pull the `internalData` properties out of my slices, and use custom hooks for them instead. I'm thinking that might be the more clean implementation.
